### PR TITLE
(BOLT-796) Document bundled module content

### PR DESCRIPTION
**Pull request by donoghuc**
This commit adds documentation for the modules packaged with bolt. Brief descriptions of module content and details for how to override bundled modules is provided.

### CODE REVIEW

File: pre-docs/bolt_installing_modules.md [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+
+Modules with useful task and plan content:
+
+- [package](https://forge.puppet.com/puppetlabs/package): Install, uninstall, update, and check the status of packages.
+- [service](https://forge.puppet.com/puppetlabs/service): Manage and inspect the state of services. 
+- [puppet_conf](https://forge.puppet.com/puppetlabs/puppet_conf): Inspect and change the configuration options in the `puppet.conf` file.
+- [facts](https://forge.puppet.com/puppetlabs/facts): Retrieve facts from specified nodes.
+- [puppet_agent](https://forge.puppet.com/puppetlabs/puppet_agent): Install Puppet Agent package.
+
+Core types and providers for applying manifest code in plans:
```
Comment (adreyer) [inline pre-docs/bolt_installing_modules.md]: I think we can just say "core puppet providers"

File: pre-docs/bolt_installing_modules.md [hunk 2]
```diff
@@ -1,6 +1,36 @@
 # Installing modules
 
-Set up Bolt to download and install modules.
+Bolt is packaged with a collection of useful modules intended to support common work flows. The included modules are described in the [Puppetfile](../Puppetfile) included in this repository. Modules and supporting documentation are publicly available on the [Puppet Forge](https://forge.puppet.com/).
```
Comment (adreyer) [inline pre-docs/bolt_installing_modules.md]: This probably shouldn't be a relative link.

File: pre-docs/bolt_installing_modules.md [hunk 3]
```diff
@@ -1,6 +1,36 @@
...
+- [scheduled_task](https://forge.puppet.com/puppetlabs/scheduled_task): Provider capable of using the more modern Version 2 Windows API for task management.
+- [sshkeys_core](https://forge.puppet.com/puppetlabs/sshkeys_core): Manage `SSH` `authorized_keys`, and `ssh_known_hosts` files. 
+- [zfs_core](https://forge.puppet.com/puppetlabs/zfs_core): Manage `zfs` and `zpool` resources.
+- [cron_core](https://forge.puppet.com/puppetlabs/cron_core): Install and manage `cron` resources.
+- [mount_core](https://forge.puppet.com/puppetlabs/mount_core): Manage mounted filesystems and mount tables.
+- [selinux_core](https://forge.puppet.com/puppetlabs/selinux_core): Manage Security-Enhanced Linux.
+- [yumrepo_core](https://forge.puppet.com/puppetlabs/yumrepo_core): Manage client yum repo configurations by parsing INI configuration files.
+- [zone_core](https://forge.puppet.com/puppetlabs/zone_core): Manage Solaris zone resources.
+
+Note that the `canary`, `aggregate` and `puppetdb_fact` modules are listed in the `Puppetfile` but are specifically implemented for bolt and not available on the forge.
```
Comment (adreyer) [inline pre-docs/bolt_installing_modules.md]: I think skip mentioning the Puppetfile here and provide a link to the directory that contains them

File: pre-docs/bolt_installing_modules.md [hunk 4]
```diff
@@ -1,6 +1,40 @@
 # Installing modules
 
-Set up Bolt to download and install modules.
+Bolt is packaged with a collection of useful modules intended to support common work flows. The included modules are described in the [Puppetfile](https://github.com/puppetlabs/bolt/blob/master/Puppetfile) included in this repository. Modules and supporting documentation are publicly available on the [Puppet Forge](https://forge.puppet.com/).
```
Comment (lucywyman) [inline pre-docs/bolt_installing_modules.md]: s/work flows/workflows/ - maybe? 
s/described/listed/ ?

File: pre-docs/bolt_installing_modules.md [hunk 5]
```diff
@@ -1,6 +1,40 @@
...
+
+**NOTE**: Bundled Forge modules are NOT PROVIDED when bolt is installed via a ruby gem. It is recommended that bolt is [installed](https://github.com/puppetlabs/bolt/blob/master/pre-docs/bolt_installing.md) as a package for most use cases.
```
Comment (lucywyman) [inline pre-docs/bolt_installing_modules.md]: @katelopresti Should this link to official docs, or does that translate to EasyDITA already?

Comment (katelopresti) [reply pre-docs/bolt_installing_modules.md]: Using internal links between markdown files is fine. [Installed](bolt_installing.md)

Comment (donoghuc) [reply pre-docs/bolt_installing_modules.md]: @katelopresti, does that mean that I need to refactor to _not_ use full `URLs` and instead use internal links? If so, does that include linking to another file in the github project (for example I want to link to the Puppetfile so the ref would look like `[Puppetfile](../Puppetfile)`?

File: pre-docs/bolt_installing_modules.md (hunk 4 above)
Comment (donoghuc) [reply pre-docs/bolt_installing_modules.md]: Will change to `workflows`.

I chose to use `described` to indicate that there is more than just a name, for example you can have a version, or information about a git repo/ref to download.

File: pre-docs/bolt_installing_modules.md (hunk 5 above)
Comment (katelopresti) [reply pre-docs/bolt_installing_modules.md]: @donoghuc Don't worry about refactoring. Full URLs are fine, just not necessary for files within the pre-docs directory. For links outside of pre-docs, like Puppetfile, a full URL is what is used in the docs published to puppet.com

**Comment by donoghuc** [comment]
Adding work for https://tickets.puppetlabs.com/browse/BOLT-820

**Comment by MikaelSmith** [comment]
BOLT-820 can be a separate PR. Just seemed related, and like a good time to address it.

**Comment by donoghuc** [comment]
I have a meeting this afternoon with Kate about this. Holding off merging if anything comes up. Thanks.

**Pull request merged**